MicrosoftDX/nether
# Simulated Analytics Players

### Issue: #486 #528

### Description:
AnalyticsTestClient has been cleaned up and functionality added. You can now:

* Generate random geo-routes throughout the world that can then be followed by virtual walking players, GeoBots.
* Simulate virtual users over a long period of time and send tens of thousands of messages to EventHub to create accurate test data.

### Test:

1. To generate routes, you first need to download the Open Street Map data file for the region where you want to generate routes for. You can download the correct file for your area [here] (http://planet.openstreetmap.org/). Save the .osm.pbf file somewhere locally on your computer.

2. Use a tool like [geojson.io](http://geojson.io) to figure out the corner coordinates for the area where you want to generate routes for.

3. Start the AnalyticsTestClient and select option 4 and then 1 to generate routes.

4. Answer the questions according to your setup and will

5. Watch how your harddrive is filled with generated routes. Open a random file and copy the content to clipboard and then paste it onto [geojson.io](http://geojson.io) to see the route on top of a map.

Keep this files on your harddrive for later use.

6. Now it's time to generate data with the routes you have. In AnalyticsTestClient, select option 4 and 2.

7. Answer the questions according to your will. Notice that quite a lot of data will be generated in a very short time, so keep the timespan short if you don't want to generate a lot of data.

8. Make sure you also start the Nether.Analytics.Host project (even though you don't need to have them running at the same time).

9. Look at the folderstructure that should be created on your harddrive as well in ADFS in the cloud.

10. Pick a random file and make sure it looks according to what you expected it to look like.

(Make sure that all above information is provided before submitting your pull request)
